src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -611,8 +624,14 @@ private Type EmitFunctionCallNode(AssociativeNode node)
             methodName = fcn.Function.Name;
             var args = fcn.FormalArguments;
             var numArgs = args.Count;
+            isBuiltIn = CoreUtils.IsInternalMethod(methodName);
```

Comment:
Why can't you call `IsInternalMethod` inside `FunctionLookup` itself? This way, you won't have to create the isBuiltIn member.

Reply:
The code currently relies on that the `className` member is stable for built in methods (it is initially null or unchanged for builtin methods). className is used in 'KeyGen' and in `Function Lookup`.

An alternative would be to make a property of 'className' and implement it as:
`internal string ClassName { get => CoreUtils.IsInternalMethod(methodName) ? nameof(BuiltIn) : className; set => className = value; }`

Do you think this approach is better?

Reply:
Ok, I got rid of the flag (without making a property of `className`).